src/controllers/resume.submission.controller.ts

```diff
@@ -31,11 +33,28 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
     if (isActiveFeature(config.FEATURE_FLAG_ENABLE_SAVE_AND_RESUME_17102022)) {
       const appData: ApplicationData = await getOverseasEntity(req, transactionId, overseaEntityId);
 
-      if (!appData || !Object.keys(appData).length) {
+      if (!Object.keys(appData || {}).length) {
         throw createAndLogErrorRequest(req, `Error on resuming OE - ${infoMsg}`);
       }
 
-      setWebApplicationData(req.session as Session, appData, transactionId, overseaEntityId);
+      const session = req.session as Session;
+      setWebApplicationData(session, appData, transactionId, overseaEntityId);
+
+      const resource = await getTransaction(req, transactionId);
+
+      // https://companieshouse.atlassian.net/wiki/spaces/TTL/pages/4063854602/Pay+Now+link+Resume
+      if (resource.status === "closed pending payment") {
+        const headersPaymentUrl = {
+          headers: {
+            [config.PAYMENT_REQUIRED_HEADER]: config.PAYMENTS_API_URL + "/payments"
```

Comment:
i'm not 100% comfortable having us maintain a payment url, one day the payment service endpoint could change and our service might break. I'll raise it with the paymentService team (Phoenix) to see if there's a way of getting the url

Reply:
Has been approved in the spike